src/vmm/src/vmm_config/hotplug.rs

```diff
@@ -0,0 +1,48 @@
+// Copyright 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::io;
+
+use serde::{Deserialize, Serialize};
+
+use crate::vstate::vcpu::VcpuError;
+use crate::StartVcpusError;
+/// Unifying enum for all types of hotplug request configs.
+/// Currently only Vcpus may be hotplugged.
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(untagged)]
+pub enum HotplugRequestConfig {
+    /// Vcpu hotplug request
+    Vcpu(HotplugVcpuConfig),
```

Comment:
Do we need the extra `HotplugVcpuConfig` type here? Wouldn't a `u8` be enough?

Reply:
I agree that it feels a bit OTT. The reason I did it this way was to utilise the serde_json `Deserialize` attribute, in-line with what all the other endpoints have done, even those with one attribute (see `MetricsConfig`, `BalloonUpdateConfig`, `BalloonUpdateStatsConfig`). Happy to change it if you think it makes more sense done alternatively.

Reply:
Ok, speaking with @roypat indeed the `u8` field won't fly due to how serde parses enums from JSON. You were right James.